### DISCUSSION TITLE
Mouse Driver

### DISCUSSION THREAD

**Discussion by GeorgeFence** [CosmosOS/Cosmos#2500] [Q&A]
Hi, i want add to my os (PatchOS) a mouse, but codes i search didn't work :(
Please can you send the code?
thanks

**Comment by zarlo** [comment]
can we see what you have tried

**Comment by GeorgeFence** [comment]
code from this video : https://www.youtube.com/watch?v=V1eRdbwExOk

**Comment by 9xbt** [comment]
try using Cosmos.System.MouseManager, it's easier to use
Getting X pos: Cosmos.System.MouseManager.X
Getting Y pos: Cosmos.System.MouseManager.Y

**Comment by AAM1075** [comment]
> try using Cosmos.System.MouseManager, it's easier to use Getting X pos: Cosmos.System.MouseManager.X Getting Y pos: Cosmos.System.MouseManager.Y

Don't forget to set the mouse sensitivity and boundaries!
Configuration for a 640x480 screen, with a mouse sensitivity of 1:
* Mouse sensitivity: `Cosmos.System.MouseManager.MouseSensitivity = 1;`
* Mouse Boundaries for width: `Cosmos.System.MouseManager.ScreenWidth = 640;`
* Mouse Boundaries for height: `Cosmos.System.MouseManager.ScreenHeight = 480;`

**Reply by 9xbt** [reply]
yeah true

**Comment by GeorgeFence** [comment]
The mouse leaves traces Do you now how to fix it?

**Comment by ghost** [comment]
canvas.Clear(Color.Black);
